packages/splits-sdk/src/types.ts

```diff
@@ -117,6 +118,174 @@ export type GetSplitBalanceConfig = {
   token?: string
 }
 
+// Warehouse
+export type WarehouseTransferConfig = {
+  receiver: Address
+  token: Address
+  amount: bigint
+} & TransactionOverridesDict
+
+export type WarehouseTransferFromConfig = {
+  sender: Address
+  receiver: Address
+  token: Address
+  amount: bigint
+} & TransactionOverridesDict
+
+export type WarehouseApproveConfig = {
+  spender: Address
+  token: Address
+  amount: bigint
+} & TransactionOverridesDict
+
+export type WarehouseSetOperatorConfig = {
+  operator: Address
+  approved: boolean
+} & TransactionOverridesDict
+
+export type WarehouseInvalidateNonceConfig = {
+  nonce: bigint
+} & TransactionOverridesDict
+
+export type WarehouseTemporaryApproveAndCallConfig = {
+  spender: Address
+  operator: boolean
+  token: Address
+  amount: bigint
+  target: Address
+  data: Hex
+} & TransactionOverridesDict
+
+export type WarehouseTemporaryApproveAndCallBySigConfig = {
+  owner: Address
+  spender: Address
+  operator: boolean
+  token: Address
+  amount: bigint
+  target: Address
+  data: Hex
+  nonce: bigint
+  deadline: number
+  signature: Hex
+} & TransactionOverridesDict
+
+export type WarehouseApproveBySig = {
+  spender: Address
+  operator: boolean
+  token: Address
+  amount: bigint
+  nonce: bigint
+  deadline: number
+}
+
+export type WarehouseTemporaryApproveAndCallBySig = {
+  spender: Address
+  operator: boolean
+  token: Address
+  amount: bigint
+  target: Address
+  data: Hex
+  nonce: bigint
+  deadline: number
+}
+
+export type WarehouseApproveBySigConfig = {
+  owner: Address
+  spender: Address
+  operator: boolean
+  token: Address
+  amount: bigint
+  nonce: bigint
+  deadline: number
+  signature: Hex
+} & TransactionOverridesDict
+
+export type WarehouseDepositConfig = {
+  receiver: Address
+  token: Address
+  amount: bigint
+} & TransactionOverridesDict
+
+export type WarehouseBatchDepositConfig = {
+  receivers: Address[]
+  token: Address
+  amounts: bigint[]
+} & TransactionOverridesDict
+
+export type WarehouseWithdrawConfig = {
+  owner: Address
+  token: Address
+} & TransactionOverridesDict
+
+export type WarehouseBatchWithdrawConfig = {
+  owner: Address
+  tokens: Address[]
+  amounts: bigint[]
+  withdrawer: Address
+} & TransactionOverridesDict
+
+export type WarehouseBatchTransferConfig = {
+  receivers: Address[]
+  token: Address
+  amounts: bigint[]
+} & TransactionOverridesDict
+
+export type WarehouseSetWithdrawConfig = {
+  incentive: number
+  paused: boolean
+} & TransactionOverridesDict
+
+export enum SplitV2Type {
+  Push,
+  Pull,
+}
+
+// Split V2
+export type CreateSplitV2Config = {
+  recipients: Address[]
+  allocations: bigint[]
+  distributionIncentive: number
```

Comment:
This was also a percent in the v1 function. Which...I think it makes sense to have it match the allocations format, so this seems ok. Should we have this be a bigint too? Just for consistency, feels funny to me to ask for bigint in the allocations but a number here. Or we can change the `allocations` to numbers. Not sure what makes more sense.

Reply:
Should definitely highlight all this in a migration guide. This feels like an easy oversight for someone moving from v1 --> v2

Reply:
the reason its not consistent with allocations is because allocations are uint256 while incentive is uint16, I think keep them separate?